src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -375,18 +375,20 @@ private SpaceType getSpaceType(final FieldInfo fieldInfo) {
 
     private boolean canDoExactSearch(final int filterIdsCount, final int searchableDocs) {
         log.debug(
-            "Info for doing exact search Live Docs: {}, filterIdsLength : {}, Threshold value: {} , Threshold %age : {}",
+            "Info for doing exact search Live Docs: {}, filterIdsLength : {}, Threshold value: {}",
             searchableDocs,
             filterIdsCount,
-            KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName()),
-            KNNSettings.getFilteredExactSearchThresholdPct(knnQuery.getIndexName())
+            KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName())
         );
+        int filterThresholdValue = KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName());
         // Refer this GitHub around more details https://github.com/opensearch-project/k-NN/issues/1049 on the logic
-        return filterIdsCount <= knnQuery.getK()
-            || (filterIdsCount <= KNNSettings.getFilteredExactSearchThreshold(knnQuery.getIndexName())
-                && (((float) filterIdsCount / (float) searchableDocs) * 100) <= (float) KNNSettings.getFilteredExactSearchThresholdPct(
-                    knnQuery.getIndexName()
-                ));
+        if (filterIdsCount <= knnQuery.getK()
```

Comment:
Just a suggestion.

```
if (filterIdsCount <= kNNQuery.getK()) {
  return true;
}

if (isThresholdSet())
  return filterThresholdValue >= filterIdsCount;
else {
  return KNNConstants.MAX_DISTANCE_COMPUTATIONS <= filterIdsCount * knnQuery.getQueryVector().length);
}



private boolean isThresholdSet() {
  return filterThresholdValue != KNNSettings.ADVANCED_FILTERED_EXACT_SEARCH_THRESHOLD_DEFAULT_VALUE;
}

```

Reply:
This is a tradeoff between readability with fewer code lines and less code branches. I choose lesser code. But I think we should prefer readability here. Let me change the code.

`isThresholdSet` I would probably move this function in the KNNSettings class, rather than making this function sit in KNNWeightClass.